src/net_processing.cpp

```diff
@@ -4732,8 +4752,8 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
                 }
 
                 // Respond to BIP35 mempool requests
-                if (fSendTrickle && pto->m_tx_relay->fSendMempool) {
-                    auto vtxinfo = m_mempool.infoAll();
+                if (fSendTrickle && pto->m_tx_relay->fSendMempool && m_mempool) {
```

Comment:
why is this check needed?

Reply:
This is indeed unnecessary due to the `!m_mempool` -> `m_tx_relay == nullptr` implication. Removing these checks.